### PR TITLE
mactex 2023.0314

### PR DESCRIPTION
**Pull request by stevenlele**
- It will take some time for all the mirrors around the world to update, see #121498

---

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Review by reitermarkus — APPROVED** [review]

**Comment by miccal** [comment]
Thank you @stevenlele.

From past experience, my local mirror at https://mirror.aarnet.edu.au/pub/CTAN/systems/mac/mactex/ is usually a good indication of when the latest MacTeX is available at almost all mirrors around the world.

So when my local tap version at https://github.com/miccal/homebrew-miccal/blob/master/Casks/m-mactex-no-gui.rb works, I will merge these two PR's.

File: Casks/mactex.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cask "mactex" do
-  version "2022.0321"
```
Comment (jb381) [inline Casks/mactex.rb:2]: 2022.0320 already available

**Comment by gromgit** [comment]
@miccal I don't know about the CTAN mirror you're using, but I can confirm that the SG mirror has 2023.0314. More importantly, it no longer hosts 2022.0321, so the current cask is broken from my POV.

I'm also not the only one affected: https://github.com/orgs/Homebrew/discussions/4352

**Comment by TangChiaHsin** [comment]
@gromgit
yep, mactex 2023.0314 is in the mirror I am using too. And sometimes the output error is `SHA mismatch`.

**Comment by bassner** [comment]
Please merge this 🙏
The current cask is broken until this is merged as it currently tries to load an old, no longer hosted version ... :( 
image

**Pull request merged**